gems/smithy/lib/smithy/util/hash_formatter.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Util
+    # A class to format a hash into a string.
+    class HashFormatter
+      # @option options [Boolean] :wrap (true) When `true`, the formatted hash will
+      #   be wrapped with curly braces.
+      # @option options [Boolean] :inline (false) When `true` the formatted hash will
+      #   contain no newlines.
+      # @option options [Boolean] :quote_strings (false) By default, all hash string values
+      #   must contain their own quotes. If you pass `true`, then all hash string
+      #   values will be inspected via `#inspect` which will auto-quote them.
+      def initialize(options = {})
+        @wrap = options.fetch(:wrap, true)
+        @inline = options.fetch(:inline, false)
+        @quote_strings = options.fetch(:quote_strings, false)
+        @indent = options.fetch(:indent, '')
+      end
+
+      # @return [String]
+      def format(obj)
+        result = hash(obj, i: @indent, inline: @inline)
+        result = unwrap(result, obj.size) unless @wrap
+        result = result.strip if @inline && result.lines.to_a.length == 1
+        result
+      end
+
+      private
+
+      def value(obj, options)
+        i = options.fetch(:i)
```

Comment:
What is `i`? It is indent? 

I find the indent slightly confusing - I slightly prefer the method we used in V3 of `levels` (eg: https://github.com/aws/aws-sdk-ruby/blob/version-3/build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/endpoint_provider_class.rb#L63).

Reply:
I changed it to indent to be consistent with everything else.